packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -1105,6 +1105,229 @@ describe('PhishingDetector', () => {
+            ],
+            name: 'test-config',
+            version: 1,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://develop.d3bkcslj57l47p.amplifyapp.com',
+          );
+          expect(result).toStrictEqual({
+            name: 'test-config',
+            result: true,
+            type: 'requestBlocklist',
+            version: '1',
+          });
+        },
+      );
+    });
+
+    it('should return false if URL is not in the requestBlocklist', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: ['hash-other.com'],
+            name: 'test-config',
+            version: 1,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://example.com',
+          );
+          expect(result).toStrictEqual({
+            result: false,
+            type: 'requestBlocklist',
+          });
+        },
+      );
+    });
+
+    it('should check all configs and return the result from the first matching config', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: [
+              '0415f1f12f07ddc4ef7e229da747c6c53a6a6474fbaf295a35d984ec0ece9455',
+            ],
+            name: 'first-config',
+            version: 1,
+            tolerance: 2,
+          },
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: [],
+            name: 'second-config',
+            version: 2,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://develop.d3bkcslj57l47p.amplifyapp.com',
+          );
+          expect(result).toStrictEqual({
+            name: 'first-config',
+            result: true,
+            type: 'requestBlocklist',
+            version: '1',
+          });
+        },
+      );
+    });
+
+    it('should return false with type "all" if no configs have a valid requestBlocklist', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            name: 'first-config',
+            version: 1,
+            tolerance: 2,
+          },
+          {
+            blocklist: [],
+            fuzzylist: [],
+            name: 'second-config',
+            version: 2,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://example.com',
+          );
+          expect(result).toStrictEqual({
+            result: false,
+            type: 'requestBlocklist',
```

Comment:
The subtest says it should be "all" but this is type 'requestBlocklist'. The current code does not support this subtest either. Is this intended?